Issue 4753 - Adjust our tests to 389-ds-base-snmp missing in RHEL 9 A…

aadhikar opened the pull request:
…ppstream

Description: With RHEL 9, 389-ds-base-snmp is no longer delivered in AppStream.
We need to adapt our tests which rely on 389-ds-base-snmp, so that they are skipped if it is missing.

Fix Description: Added skipif to tests which rely on 389-ds-base-snmp

Fixes: https://github.com/389ds/389-ds-base/issues/4753

Reviewed by: ???

aadhikar committed:
Issue 4753 - Adjust our tests to 389-ds-base-snmp missing in RHEL 9 Appstream

Description: With RHEL 9, 389-ds-base-snmp is no longer delivered in AppStream.
We need to adapt our tests which rely on 389-ds-base-snmp, so that they are skipped if it is missing.

Fix Description: Added skipif to tests which rely on 389-ds-base-snmp

Fixes: https://github.com/389ds/389-ds-base/issues/4753

Reviewed by: ??

vashirov commented on src/lib389/lib389/utils.py:
```diff
@@ -1431,3 +1431,12 @@ def is_valid_hostname(hostname):
     return all(allowed.match(x) for x in hostname.split("."))
+
+
+def snmp_present():
```
I don't think this should be in lib389.
We have `get_rpm_version()` in `conftest.py`: https://github.com/389ds/389-ds-base/blob/a79433498709319814b7dae93c3dec9a0ed32c5c/dirsrvtests/conftest.py#L27-L34
It also won't fail on the systems where `rpm` is not present.

aadhikar committed:
removed the snmp_present() from utils.py as we have get_rpm_version() in conftest.py

sgouvern reviewed (approved):
Minor remark : the reason why the test is skipped ("not present") does not appear in the test output.
Apart from that, ok for me

sgouvern reviewed (commented):
My bad, indeed the reason is displayed : (Not present). 
But could you please elaborate a bit more so that the reason for skipping appears clearly, like '389-ds-base-snmp package is not present' ?

aadhikar committed:
Issue 4753 - Adjust our tests to 389-ds-base-snmp missing in RHEL 9 Appstream

Description: With RHEL 9, 389-ds-base-snmp is no longer delivered in AppStream.
We need to adapt our tests which rely on 389-ds-base-snmp so that they are skipped if it is missing.

Fix Description: Added skipif to tests that rely on 389-ds-base-snmp

Fixes: https://github.com/389ds/389-ds-base/issues/4753

Reviewed by: @vashirov, @sgouvern (Thanks!)

sgouvern reviewed (commented):
It's nitpicking, but the reason should be "389-ds-base-snmp package is ***not*** present" :)

sgouvern reviewed (approved):
Thanks :)
LGTM !
